File: client-app/src/app/models/activity.ts
import { act } from "react-dom/test-utils";
import { Profile } from "./profile";

export interface Activity {
    id:          string;
    title:       string;
    date:        Date | null;
    description: string;
    category:    string;
    city:        string;
    venue:       string;
    hostUsername: string;
    isCancelled: boolean;
    isGoing: boolean;
    isHost: boolean;
    host?: Profile;
    attendees: Profile[];
}

//將ActivityFormValues資料轉換到Activity
export class Activity implements Activity{
    constructor(init?: ActivityFormValues) {
        Object.assign(this, init)
    }
}

//將從API得到的Activity資料，轉換成ActivityForm型態
export class ActivityFormValues{
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date: Date | null = null;
    city: string = '';
    venue: string = '';

    constructor(activity?: ActivityFormValues) {
        //已有activity => 修改
        if(activity){
            this.id = activity.id;
            this.title = activity.title;
            this.category = activity.category;
            this.description = activity.description;
            this.date = activity.date;
            this.city = activity.city;
            this.venue = activity.venue;
        }
    }
}

File: Domain/Activity.cs
using System;

namespace Domain
{
    public class Activity
    {
        public Guid Id { get; set; }

        public string Title { get; set; }

        public DateTime Date { get; set; }

        public string Description { get; set; }

        public string Category { get; set; }

        public string City { get; set; }

        public string Venue { get; set; }

        public bool IsCancelled { get; set; }

        //新增Activity時，EntityFrameWork不會幫我們新增Attendee
        //Attendee會是null，此時無法給值 ==> 給予Attendee初始值，讓Attendee為一空陣列
        public ICollection<ActivityAttendee> Attendees { get; set; } = new List<ActivityAttendee>();

        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}